app/models/service_order.rb

```diff
@@ -24,7 +26,10 @@ def create_order_name
     update_attributes(:name => "Order # #{id}") if name.blank?
   end
 
-  def checkout
+  def checkout(checkout_immediately = false)
+    unless checkout_immediately
+      raise "Invalid operation [checkout] for Service Order in state [#{state}]" if state == STATE_ORDERED
```

Comment:
I think this would be simpler to read like this (I'm not a fan of `unless`):
```ruby
raise "Invalid operation [checkout] for Service Order in state [#{state}]" if !checkout_immediately && state == STATE_ORDERED
```